plugins/vst_base/RemoteVstPlugin.cpp

```diff
@@ -1807,6 +1851,13 @@ DWORD WINAPI RemoteVstPlugin::guiEventLoop( LPVOID _param )
 	while( quit == false && GetMessage( &msg, NULL, 0, 0 ) )
 	{
 		TranslateMessage( &msg );
+
+		if( msg.message == WM_SYSCOMMAND && msg.wParam == SC_CLOSE )
+		{
+			_this->destroyEditor();
```

Comment:
@lukas-w When I merged this to my local branch(up-to-date `stable-1.2`) it fails to build. The failure is because of fe98a9a by @DomClark. It seems to work after I change `_this` to `__plugin`, but I'm not sure this is the right way...

Reply:
Previously `_this` was passed as an argument to `guiEventLoop`, but after I moved the window message handling code to `messageWndProc`, there was no need for it any more so I removed it. Using `__plugin` is safe here, and has the same value `_this` would have done, so the change is correct, but this code may well be better off with all the other message handling code in `messageWndProc`. I have no experience with Linux programming, so I may well be wrong, but on Windows other modal loops can be run, e.g. by modal dialogs or window drag-resizing, in which case the messages are still sent to the WindowProc but will not be picked up by code in the main modal loop. This could lead to your message being missed if it is dispatched while a different loop is currently in control, unless the code is moved over.

Reply:
@DomClark Thanks for explanation. :+1: